packages/react-native/React/DevSupport/RCTInspectorDevServerHelper.mm

```diff
@@ -50,17 +128,22 @@
 #endif // macOS]
   NSString *escapedAppName = [[[NSBundle mainBundle] bundleIdentifier]
       stringByAddingPercentEncodingWithAllowedCharacters:NSCharacterSet.URLQueryAllowedCharacterSet];
-  return [NSURL URLWithString:[NSString stringWithFormat:@"http://%@/inspector/device?name=%@&app=%@",
+
+  NSString *escapedInspectorDeviceId = [getInspectorDeviceId()
+      stringByAddingPercentEncodingWithAllowedCharacters:NSCharacterSet.URLQueryAllowedCharacterSet];
+
+  return [NSURL URLWithString:[NSString stringWithFormat:@"http://%@/inspector/device?name=%@&app=%@&device=%@",
```

Comment:
nit: does this need a [macOS] tag?

Reply:
I just did a diff with React Native upstream. There was one tag missing, but not here. I'll add it in a commit, and then rebase before merge so we can keep the number of commits low